bank.rb

```diff
@@ -0,0 +1,39 @@
+require_relative 'account'
+# SauceBank
+class Bank
+  attr_accessor :accounts
+  def initialize
+    @accounts ||= load_accounts
+  end
+
+  def load_accounts
+    []
+  end
+
+  def find_account(name)
+    @accounts.select do |acct|
+      acct.owner.name.downcase == name.downcase
+    end
+  end
+
+  def has_account?(name)
+    query = find_account(name)
+    !query.empty?
+  end
+
+  def create_account(name)
+    new_acct = Account.new({ owner: name })
+    add_account(new_acct)
+    new_acct
+  end
+
+  def add_account(acct)
+    puts 'account added.'
```

Comment:
The bank shouldn't be responsible for communicating with the user.  This would be better in a UI class.

This also makes the code more reusable.  What if you wanted to bundle your code as a gem, and plug it into a rails app? The view wouldn't pull data from the bank to communicate to your user.

Reply:
Ok! I'll pull those puts statements